crypto/s2n_hmac.c

```diff
@@ -44,38 +62,38 @@ int s2n_hmac_hash_alg(s2n_hmac_algorithm hmac_alg, s2n_hash_algorithm *out)
     default:
         S2N_ERROR(S2N_ERR_HMAC_INVALID_ALGORITHM);
     }
-    return 0;
+    return S2N_SUCCESS;
 }
 
 int s2n_hmac_digest_size(s2n_hmac_algorithm hmac_alg, uint8_t *out)
 {
     s2n_hash_algorithm hash_alg;
     GUARD(s2n_hmac_hash_alg(hmac_alg, &hash_alg));
     GUARD(s2n_hash_digest_size(hash_alg, out));
-    return 0;
+    return S2N_SUCCESS;
 }
 
 /* Return 1 if hmac algorithm is available, 0 otherwise. */
-int s2n_hmac_is_available(s2n_hmac_algorithm hmac_alg)
+bool s2n_hmac_is_available(s2n_hmac_algorithm hmac_alg)
 {
-    int is_available = 0;
+    bool is_available = false;
     switch(hmac_alg) {
     case S2N_HMAC_MD5:
     case S2N_HMAC_SSLv3_MD5:
     case S2N_HMAC_SSLv3_SHA1:
         /* Set is_available to 0 if in FIPS mode, as MD5/SSLv3 algs are not available in FIPS mode. */
         is_available = !s2n_is_in_fips_mode();
         break;
-    case S2N_HMAC_NONE:    
+    case S2N_HMAC_NONE:
     case S2N_HMAC_SHA1:
     case S2N_HMAC_SHA224:
     case S2N_HMAC_SHA256:
     case S2N_HMAC_SHA384:
     case S2N_HMAC_SHA512:
-        is_available = 1;
+        is_available = true;
         break;
     default:
-        S2N_ERROR(S2N_ERR_HMAC_INVALID_ALGORITHM);
+        S2N_ERROR_BOOL(S2N_ERR_HMAC_INVALID_ALGORITHM);
```

Comment:
This seems weird to me. Functions that return bool shouldn't set the errno, as the caller has no way of knowing if `false` means it's not available or it means that it supplied a bad arg. We should either remove this or return a `S2N_RESULT` and have an out param.

Reply:
I've refactored `s2n_hmac_is_available` similarly to `s2n_hash_is_available`, so this is no longer necessary.